src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
```

Comment:
[Make node an inner class and make constructor in it like in LL class](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#make-node-an-inner-class)

Reply:
Hopefully I right understood your mean

```java
public class MyLinkedList<T> implements MyLinkedListInterface<T> {
    private static final String INDEX_LESS_THAN_ZERO_MESSAGE = "Index can't be less than zero";
    private static final String INDEX_NOT_FOUND_MESSAGE = "Not found index: ";
    private int size;
    private Node<T> head;
    private Node<T> tail;

    private static class Node<T> {
        private T item;
        private Node<T> next;
        private Node<T> prev;

        public Node(Node<T> prev, T element, Node<T> next) {
            this.item = element;
            this.next = next;
            this.prev = prev;
        }
    }
```